### PR TITLE
Set `NONE` Type Descriptor Kind for `NONE` and `OTHER` Type Kinds

### LINKED ISSUE #33666: Null as the TypeKind is captured
**Description:**
Consider the following source snippet

```
function getIntVal() {
    worker w1 {
        int x = 12;
        x -> w2;
    }

    worker w2 {
        <-
    }
}
```

**Steps to reproduce:**
Within the worker `w2` we have an incomplete worker receive action where the diagnostic properties contains the type symbols. If we get the type kind of the type symbol we get a null. Instead we should have `NONE` or `CompilationError` type kind. This is resulting of NPEs when using the API.

In order to observe the behaviour, get the cursor to the particular erroneous location and you can see the following stack trace thrown from the code actions.

```
Error - 4:40:35 PM] CodeAction 'CreateVariableCodeAction' failed! {, error: 'java.lang.NullPointerException'}
java.lang.NullPointerException
	at org.ballerinalang.langserver.common.utils.CommonUtil.generateVariableName(CommonUtil.java:780)
	at org.ballerinalang.langserver.codeaction.providers.createvar.CreateVariableCodeAction.getCreateVariableTextEdits(CreateVariableCodeAction.java:112)
	at org.ballerinalang.langserver.codeaction.providers.createvar.CreateVariableCodeAction.getDiagBasedCodeActions(CreateVariableCodeAction.java:77)
	at org.ballerinalang.langserver.codeaction.CodeActionRouter.lambda$getAvailableCodeActions$4(CodeActionRouter.java:109)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
	at org.ballerinalang.langserver.codeaction.CodeActionRouter.lambda$getAvailableCodeActions$5(CodeActionRouter.java:103)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at org.ballerinalang.langserver.codeaction.CodeActionRouter.getAvailableCodeActions(CodeActionRouter.java:101)
	at org.ballerinalang.langserver.codeaction.BallerinaCodeActionExtension.execute(BallerinaCodeActionExtension.java:46)
	at org.ballerinalang.langserver.codeaction.BallerinaCodeActionExtension.execute(BallerinaCodeActionExtension.java:34)
	at org.ballerinalang.langserver.LangExtensionDelegator.codeActions(LangExtensionDelegator.java:157)
	at org.ballerinalang.langserver.BallerinaTextDocumentService.lambda$codeAction$8(BallerinaTextDocumentService.java:328)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:642)
	at java.base/java.util.concurrent.CompletableFuture$Completion.exec(CompletableFuture.java:479)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
```

**Affected Versions:**
SwanLake-beta3 at least

### LINKED ISSUE #34031: Logical error in the `withinRange()` method of the Semantic API
**Description:**
The `withinRange(LineRange range, LineRange specifiedRange)` method in the `BallerinaSemanticModel` would return `false` for the following example where it is supposed to return `true`:

```
LineRange lineRange = new LineRange("file.bal", new LinePosition(2, 1), new LinePosition(3, 2));
LineRange range = new LineRange("file.bal", new LinePosition(1, 0), new LinePosition(4, 0)) ;
withinRange(lineRange, range)
```

**Steps to reproduce:**

**Affected Versions:**

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

### PR DESCRIPTION
**Pull request by sanjana**
Fixes #33666
Fixes #34031

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/BallerinaSemanticModel.java [hunk 1]
```diff
@@ -458,14 +458,17 @@ private BPackageSymbol getModuleSymbol(BLangCompilationUnit compilationUnit) {
     private boolean withinRange(LineRange range, LineRange specifiedRange) {
```
Comment (pubudu91) [inline compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/BallerinaSemanticModel.java:458]: Shall we look at getting rid of this method and making use of the `withinRange()` in `PositionUtil`? Don't think this method is properly implemented.

File: language-server/modules/langserver-core/src/test/resources/completion/listener_decl/config/config8.json [hunk 2]
```diff
@@ -8,7 +8,7 @@
     {
       "label": "test",
       "kind": "Variable",
-      "detail": "none",
+      "detail": "$UndefinedType$",
```
Comment (IMS94) [inline language-server/modules/langserver-core/src/test/resources/completion/listener_decl/config/config8.json:11]: Shall we create an issue under LS label to address `$UndefinedType$` being set as the detail. A minor change is needed at `AbstractCompletionProvider.java:193`.

**Review by IMS94 — APPROVED** [review]
LGTM! Let's create an LS side issue as I have commented.

**Comment by mohanvive** [comment]
@sanjana can you please attend these comments.

File: compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/BallerinaSemanticModel.java (hunk 1 above)
Comment (sanjana) [reply compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/BallerinaSemanticModel.java]: @pubudu91, Moved with the new logic. The existing method which was already in the `PositionUtil` was renamed to `isRangeWithinNode()`.

**Review by mohanvive — APPROVED** [review]

File: compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/PositionUtil.java [hunk 3]
```diff
@@ -59,45 +59,26 @@ static boolean withinBlock(LinePosition cursorPos, Location symbolPosition) {
...
+
+        int enclStartLine = enclRange.startLine().line();
+        int enclEndLine = enclRange.endLine().line();
+        int enclStartOffset = enclRange.startLine().offset();
+        int enclEndOffset = enclRange.endLine().offset();
+
+        return enclRange.filePath().equals(range.filePath())
+                && (startLine == enclStartLine && startOffset >= enclStartOffset || startLine > enclStartLine)
+                && (endLine == enclEndLine && endOffset <= enclEndOffset || endLine < enclEndLine);
+    }
```
Comment (pubudu91) [inline compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/PositionUtil.java:76]: Had a closer look at this. Good job on this simplification!

**Pull request merged**